config/routes.rb

```diff
@@ -119,7 +119,12 @@ def scoped_navigation_routes(context, navigation)
       get "/relational-efin", to: "consent_pages#relational_efin"
       get "/global-carryforward", to: "consent_pages#global_carryforward"
 
-      devise_for :clients, skip: [:sessions]
```

Comment:
`devise_for :client`, in addition to (sometimes) writing routes, primary mounts devise as an engine (provides current_client functionality, etc). This writing of the helpers (and whatever else the engine is doing) seems to only need to happen once, not once per subdomain (which is why current_client was working even when we didn’t have devise_for defined under the :ctc subdomain.) 

In fact, I think we can continue to define devise_for :clients just once, and potentially even move it to the top of the routes file so that it’s clear it doesn’t “belong” solely to either subdomain.

If you check the docs for devise_for, it does also add some routes sometimes:
https://www.rubydoc.info/github/plataformatec/devise/ActionDispatch%2FRouting%2FMapper:devise_for

However, it only adds routes when you’re using certain supported modules. For example, it adds the sessions routes when :authenticatable is a devise module defined on Client. Since we don’t use that module, we can remove `skip: :sessions` since those routes won’t be written anyway. (You can test this out by toggling adding/removing `skip :sessions` and noticing that there are no additional routes written when removing.

See the difference between the devise modules used on the two Devise-powered classes: Client line 44 vs User line 51

Reply:
I also see that we changed devise_for to use the singular :client instead of plural :clients. That makes logical sense to me given that devise_scope needs the singular :client, but while snooping around the docs (clicking on devise_for in Rubymine), I found this:

# Also be aware of that 'devise_scope' and 'as' use the singular form of the
# noun where other devise route commands expect the plural form. 

It’s possible that changing devise_for back to plural :clients could maybe save some random pain in the future, it is also possible that it will never affect us whatsoever!